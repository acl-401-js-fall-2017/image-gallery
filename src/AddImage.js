import React, { PureComponent } from 'react';

export default function AddImage ({ handleOnSubmit }) {
  
    return (
        <div>
          <h2>Add a new wonder!</h2>
          <form onSubmit={event => {
              event.preventDefault();
              const { elements } = event.target;
              handleOnSubmit(elements.title.value, elements.description.value, elements.url.value);
          }}>
            <input type="text" name="title" placeholder="wonder title" required />
            <br/>
            <textarea rows="5" cols="60" name="description" placeholder="wonder description" required />
            <br/>
            <input type="url" name="url" placeholder="wonder image url" required />
            <br/>
            <button type="submit">Save</button>
          </form>
        </div>
      );
    
    }
  
  