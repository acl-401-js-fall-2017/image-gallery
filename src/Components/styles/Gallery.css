.Gallery {
    height: 100%;
}

.imgSlide {
    height: 60%;
    width: 80%;
    margin: 1em auto;
    overflow: hidden;
}

.Gallery img {
    height: 90%;

}

.galleryButtons {
    display: flex;
    justify-content: center;
    height: 10%;
}

.galleryButtons h1 {
    width:50%;
    text-align: center;
    background-color: black;
    border-radius: 0.25em;
    color: white;
    margin: 0.75em 1em;
    line-height: 1.5em;
}