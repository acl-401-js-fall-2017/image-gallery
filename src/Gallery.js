import React, { PureComponent } from 'react';
import styled from 'styled-components';

export default class Thumbnail extends PureComponent {
  constructor() {
    super();
    this.state= { index: 0}
    this.history =[];
  }

  setState(state, ignore) {
    if(!ignore) this.history.push(state);
    super.setState(state);
  }

  handleClick = (value) =>{
    console.log('clicked');
    const newState ={
      ...this.state,
      index: this.state.index + value
    }
    this.setState(newState);
  }
  render() {
    const {gallery, shouldDisplay } =this.props
    return( 
      <StyledDiv shouldDisplay={shouldDisplay}>
        <div>
          {this.props.gallery.map((img, i) => (
            <ImgDiv shouldDisplay ={this.state.index === i}>
              <span> {img.title} </span>
              <img style={{ width:'100%' }} key={img._id} src={img.img} alt='image'/>
              <span> {img.description} </span>
              { i !== gallery.length -1 && <span onClick ={()=> this.handleClick(1)}> next </span>}
              {i !== 0 && <span onClick ={()=> this.handleClick(-1)}> previous</span>}
            </ImgDiv>
          ))}
        </div>
      </StyledDiv>
    )
  };
}

const StyledDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
`;

const ImgDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
flex-direction: column;
margin: 0 10%;
`;

