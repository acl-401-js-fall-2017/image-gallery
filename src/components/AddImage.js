import React, { Component } from  'react';
import shortid from 'shortid';

export default class AddImage extends Component {

  render() {

    const { addImage } = this.props;

    return (
      <div>
        <form onSubmit={event => {
          event.preventDefault();
          const { elements } = event.target;
          const newImage = {
            _id: shortid.generate(),
            title: elements.title.value,
            description: elements.description.value,
            url: elements.url.value
          };
          elements.title.value = elements.description.value = elements.url.value = '';
          return addImage(newImage);
        }}>
          <input name="title" placeholder="Title" />
          <input name="description" placeholder="Description" />
          <input name="url" placeholder="Url" />
          <button type="submit"> Add Image </button>
        </form>
      </div>
    );
  }
}