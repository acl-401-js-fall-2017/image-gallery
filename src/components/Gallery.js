import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Gallery extends PureComponent {
  constructor(){
    super();
    this.state = {
      index: 0
    };
  }
  handleImageSelection(value){
    let i = value === 'next' ? 1: -1;
    this.setState({ index: this.state.index + i });
  }
  render(){
    const { images } = this.props;
    const limit = this.state.index === images.length - 1; 
    const currentImage = images.length ? images[this.state.index] : null;
    const display = currentImage ? <img src={currentImage.url} alt={currentImage.description} height="40%" width="40%"/> : null;
    const next = !limit ? <input type="button" value="next" onClick={({ target }) => this.handleImageSelection(target.value)}/> : null;
    const back = this.state.index !== 0 ? <input type="button" value="back" onClick={({ target }) => this.handleImageSelection(target.value)}/> : null;
    return(
      <div>
        {next}
        {back}
        <br/>
        {display}
      </div>
    );
  }
}
Gallery.propTypes = {
  images: PropTypes.array
};

