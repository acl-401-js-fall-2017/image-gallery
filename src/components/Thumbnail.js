import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Thumbnail extends PureComponent {
  render(){
    const { images } =this.props;
    const list = images.map((image, index)=> {
      const listItem =
        <li key={index} float="left">
          <figcaption><a href={image.url}>{image.title}:</a></figcaption>
          <img src={image.url} alt={image.url} height="100" width="100"/>
        </li>;
      return listItem;
    });
    return(
      <ul>
        {list}
      </ul>
    );
  }
}

Thumbnail.propTypes = {
  images: PropTypes.array
};