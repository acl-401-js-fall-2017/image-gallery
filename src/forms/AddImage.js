import React, { PureComponent } from 'react';
    
export default class AddImage extends PureComponent {
  render() {
    const { handleOnSubmit } = this.props;
    return (
      <div>
        <h2>Add a new image!</h2>
        <form onSubmit={event => {
          event.preventDefault();
          const { elements } = event.target;
          handleOnSubmit(elements.title.value, elements.description.value, elements.url.value);
        }}>
          <input type="text" name="title" placeholder="image title" required />
          <br/>
          <textarea rows="5" cols="60" name="description" placeholder="image description" required />
          <br/>
          <input type="url" name="url" placeholder="image url" required />
          <br/>
          <button type="submit">Save</button>
        </form>
      </div>
    );
  }
}