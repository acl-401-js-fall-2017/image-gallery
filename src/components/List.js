import React, { Component } from 'react';
import PropTypes from 'prop-types';

import AddImage from './AddImage';

export default class List extends Component {
   
    static propTypes = {
      images: PropTypes.arrayOf(PropTypes.object).isRequired
    }


    render() {
      const { addImage, handleRemove } = this.props;
      return (
        <div className="list-view">
          <table>
            <tbody>
              {this.props.images.map((image, index) => {
                return (
                  <tr key={index}>
                    <td><a href={image.url}>{image.title}</a></td>
                    <td>{image.description}</td>
                    <td> <input type="button" data-value={image._id} value="remove" onClick={() => handleRemove(image._id)}/></td>
                  </tr>
                );
              })}
              <tr>
                <td></td>
                <td><AddImage addImage={newImage => addImage(newImage)}/></td>
                <td></td>
              </tr>
            </tbody>
          </table>
        </div>
      );
    }
}