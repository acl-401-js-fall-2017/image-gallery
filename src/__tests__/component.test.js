import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AddImage from '../components/add-image'; 
import Gallery from '../components/Gallery'; 
import List from '../components/List';
import Thumbnail from '../components/Thumbnail';


const images =[
  {
    _id: 1,
    title: 'Cute Bunny',
    description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
    url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
  },
  { 
    _id: 2,
    title: 'Cute Bunny2',
    description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg'
  },
  { 
    _id: 3,
    title: 'Cute Bunny3',
    description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg'
  }
];

describe('components', () => {

 
    
  it('should render AddImage component', () => {
    const wrapper = shallow(<AddImage/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render Gallery component', () => {
    const wrapper = shallow(<Gallery images={images}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render List component', () => {
    const wrapper = shallow(<List images={images}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render Thumbnail component', () => {
    const wrapper = shallow(<Thumbnail images={images}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});