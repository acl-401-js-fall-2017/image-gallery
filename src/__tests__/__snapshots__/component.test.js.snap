// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components should render AddImage component 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Title: 
  </label>
  <input
    name="title"
    type="text"
  />
  <br />
  <label>
    URL: 
  </label>
  <input
    name="url"
    type="text"
  />
  <br />
  <label>
    Description: 
  </label>
  <input
    name="description"
    type="text"
  />
  <br />
  <button
    type="submit"
  >
    Add
  </button>
</form>
`;

exports[`components should render Gallery component 1`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="next"
  />
  <br />
  <img
    alt="Isn't it fuzzy-wuzzy cutest thing you've ever seen?"
    height="40%"
    src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
    width="40%"
  />
</div>
`;

exports[`components should render List component 1`] = `
<div>
  <table>
    <tbody>
      <tr
        key="0"
      >
        <td>
          <a
            href="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
          >
            Cute Bunny
          </a>
        </td>
        <td>
          <input
            data-value={1}
            onClick={[Function]}
            type="button"
            value="X"
          />
        </td>
      </tr>
      <tr
        key="1"
      >
        <td>
          <a
            href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
          >
            Cute Bunny2
          </a>
        </td>
        <td>
          <input
            data-value={2}
            onClick={[Function]}
            type="button"
            value="X"
          />
        </td>
      </tr>
      <tr
        key="2"
      >
        <td>
          <a
            href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
          >
            Cute Bunny3
          </a>
        </td>
        <td>
          <input
            data-value={3}
            onClick={[Function]}
            type="button"
            value="X"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <AddImage />
</div>
`;

exports[`components should render Thumbnail component 1`] = `
<ul>
  <li
    float="left"
    key="0"
  >
    <figcaption>
      <a
        href="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
      >
        Cute Bunny
        :
      </a>
    </figcaption>
    <img
      alt="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
      height="100"
      src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
      width="100"
    />
  </li>
  <li
    float="left"
    key="1"
  >
    <figcaption>
      <a
        href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
      >
        Cute Bunny2
        :
      </a>
    </figcaption>
    <img
      alt="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
      height="100"
      src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
      width="100"
    />
  </li>
  <li
    float="left"
    key="2"
  >
    <figcaption>
      <a
        href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
      >
        Cute Bunny3
        :
      </a>
    </figcaption>
    <img
      alt="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
      height="100"
      src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
      width="100"
    />
  </li>
</ul>
`;
